identity/src/commonMain/kotlin/com/android/identity/util/ByteArrayUtil.kt

```diff
@@ -0,0 +1,126 @@
+package com.android.identity.util
+
+/**
+ * ByteArray extension functions for consistent reading and writing commonly used types.
+ */
+fun ByteArray.putInt8(value: Int, offset: Int = 0) {
+    require(value in Byte.MIN_VALUE..Byte.MAX_VALUE) { "Value $value is out of Int8 range" }
+    this[offset] = value.toByte()
+}
+
+fun ByteArray.putInt16(value: Int, offset: Int = 0) {
+    require(value in Short.MIN_VALUE..Short.MAX_VALUE) { "Value $value is out of Int16 range" }
+    this[offset] = (value and 0xFF).toByte() // Lower byte.
+    this[offset + 1] = ((value shr 8) and 0xFF).toByte()  // Higher byte.
+}
+
+fun ByteArray.putInt32(value: Int, offset: Int = 0) {
+    this[offset] = (value and 0xFF).toByte()
+    this[offset + 1] = ((value shr 8) and 0xFF).toByte()
+    this[offset + 2] = ((value shr 16) and 0xFF).toByte()
+    this[offset + 3] = ((value shr 24) and 0xFF).toByte()
+}
+
+fun ByteArray.putInt64(value: Long, offset: Int = 0) {
```

Comment:
It would look a lot more natural if the offset is the first parameter and the value is the second. I also don't think we should use a default value for the offset, I think we want to nudge the developer to remember they need to specify it.